modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobStore.java

```diff
@@ -120,9 +124,22 @@ class S3BlobStore implements BlobStore {
     RequestMetricCollector getMetricCollector(Operation operation, OperationPurpose purpose) {
         var collector = statsCollectors.getMetricCollector(operation, purpose);
         return new RequestMetricCollector() {
+
+            private final Map<String, Object> attributes = Map.of(
+                "repo_type",
+                S3Repository.TYPE,
+                "repo_name",
+                repositoryMetadata.name(),
+                "operation",
+                operation.getKey(),
+                "purpose",
+                purpose.getKey()
+            );
+
             @Override
             public void collectMetrics(Request<?> request, Response<?> response) {
                 s3RequestRetryStats.addRequest(request);
+                requestCounter.incrementBy(getRequestCount(request), attributes);
```

Comment:
:+1: 

Is it a right time to start discussing additional metrics or is it postponed for the future prs?

Reply:
Also is it possible to integrate this counter into collector on the following line?

Reply:
We can start discussing additional metrics. For this PR though, I'd like to keep it isolated because it serves as a simple example on how we want to tackle this and also having it in the serverless build earlier for potential manual test there which does not seem to work at this stage. But that means even more reasons to get at least one metric in earlier.

> Also is it possible to integrate this counter into collector on the following line?

Yeah I can do that. It was easier for me to directly add it here. Since this annoymous class may be temporary, I think you are right about intergrating into the collector.